Randomizer.App/TrackerLocationSyncer.cs

```diff
@@ -0,0 +1,191 @@
+
+        public IEnumerable<Location> AllClearableLocations => World.Locations.Where(x => IsLocationClearable(x)).ToImmutableList();
+
+        /// <summary>
+        /// Returns the regions sorted by the ones with highest number of accessible items, stickying the one that the user recently
+        /// </summary>
+        /// <param name="filter">LocationFilter for SM or Z3</param>
+        /// <param name="applyStickyRegion">If the region the player updated last should be stickied at the top</param>
+        /// <returns>A list of all regions sorted by the regions with the most items</returns>
+        public IEnumerable<Region> GetTopRegions(LocationFilter filter = LocationFilter.None, bool applyStickyRegion = true)
+        {
+            var regions = World.Regions
+                .Where(x => RegionMatchesFilter(filter, x))
+                .OrderByDescending(x => x.Locations.Count(x => !x.Cleared && x.IsAvailable(Progression)))
+                .ToList();
+
+            if (applyStickyRegion && _stickyRegion != null && regions.Contains(_stickyRegion))
+                regions.MoveToTop(x => x == _stickyRegion);
+
+            return regions;
+        }
+
+        /// <summary>
+        /// Returns the locations sorted by regions with the highest number of accessible items
+        /// </summary>
+        /// <param name="filter">LocationFilter for SM or Z3</param>
+        /// <param name="applyStickyRegion">If the region the player updated last should be sticked at the top</param>
+        /// <returns>A list of all locations sorted by the regions with the most items</returns>
+        public IEnumerable<Location> GetTopLocations(LocationFilter filter = LocationFilter.None, bool applyStickyRegion = true)
+        {
+            return GetTopRegions(filter, applyStickyRegion).SelectMany(x => x.Locations)
+                .Where(x => IsLocationClearable(x))
+                .ToImmutableList();
+        }
+
+        /// <summary>
+        /// Returns if a SMZ3 location is currently accessible
+        /// </summary>
+        /// <param name="location">The location to check</param>
+        /// <param name="allowOutOfLogic">If out of logic checks shoudl be returned, assuming the option is enabled</param>
+        /// <returns>True if a location is accessible given settings, false otherwise</returns>
+        public bool IsLocationClearable(Location location, bool allowOutOfLogic = true)
+        {
+            return !location.Cleared && (location.IsAvailable(Progression) || location.IsAvailable(ProgressionWithKeys) || (allowOutOfLogic && ShowOutOfLogicLocations));
+        }
+
+        /// <summary>
+        /// Clears a location and fires the updated events
+        /// </summary>
+        /// <param name="location">The SMZ3 location to update</param>
+        public void ClearLocation(Location location)
+        {
+            location.Cleared = true;
```

Comment:
I agree about not wanting to introduce too many layers. I mostly just figured it made sense here because this reduces duplicated code that could introduce inconsistencies between the locations list and map. Now, admittedly, I possibly could have just added a lot of this to the tracker itself.

For the Tracker.Clear, I had noticed that you didn't previously call it for the x button, so I pretty much duplicated the behavior here to remain consistent. I had assumed it was maybe because if people were tracking things by hand they may not want extra stuff, but I updated it. The undo functionality alone is worth it, speaking as someone who accidentally tracked something by hand wrong last night and couldn't undo it. 😂

Reply:
Yeah, I think tracker might be the right place for some of this, but that's probably a somewhat bigger undertaking, trying to untangle some of the code there already.

> For the Tracker.Clear, I had noticed that you didn't previously call it for the x button, so I pretty much duplicated the behavior here to remain consistent.
Yeah, that was _me_ being inconsistent. I just never used the button to clear locations much so it never occurred to me. I should probably update that at some point.

It's definitely worth looking at Tracker.cs at some point. TrackItem is called directly from the voice command so does everything it needs to... but ended up being called by other voice commands, so it gets a bit messy. Could use some restructuring, but I'm not sure how exactly yet.

Reply:
> Yeah, that was _me_ being inconsistent. I just never used the button to clear locations much so it never occurred to me. I should probably update that at some point.

Since I made the syncer update, this will update that as well as I had updated the x button to use the syncer.

But, yeah, I probably need to do a deeper dive. I had originally been approaching this from a perspective of not wanting to step on toes so to speak and keep my stuff from interfering with yours, but I feel like some things could have been done better on my side from taking things at a more wholistic approach, so before doing much more I'll try to go through the Tracker's codebase more.